src/icon-manifest/iconGenerator.ts

```diff
@@ -290,20 +288,27 @@ export class IconGenerator implements models.IIconGenerator {
 
   private hasCustomIcon(customIconFolderPath: string, filename: string): boolean {
     const relativePath = utils.getRelativePath('.', customIconFolderPath, false);
-    const filePath = path.posix.join(relativePath, filename);
+    const filePath = utils.pathUnixJoin(relativePath, filename);
     return fs.existsSync(filePath);
   }
 
-  private getIconPath(defaultPath: string, filename: string): string {
-    const absPath = path.join(this.settings.vscodeAppData, this.settings.extensionSettings.customIconFolderName);
-    if (!this.avoidCustomDetection && this.hasCustomIcon(absPath, filename)) {
-      const sanitizedFolderPath =
-        utils.belongToSameDrive(absPath, this.manifestFolderPath)
-          ? this.manifestFolderPath
-          : utils.overwriteDrive(absPath, this.manifestFolderPath);
-      return utils.getRelativePath(sanitizedFolderPath, absPath, false);
+  private getIconPath(filename: string): string {
+    if (this.avoidCustomDetection) {
+      return this.iconsFolderBasePath;
+    }
+    const customIconFolderPath = this.configCustomIconFolderPath && this.configCustomIconFolderPath !== ''
```

Comment:
The sencond statement is redundant as `''`' is a falsy value which is already checked in the first statement.